hack/generated/Taskfile.yml

```diff
@@ -0,0 +1,113 @@
+  # the code has not been go fmt-ed, whereas locally we want "make all"
+  # to just format the code for you
+  ci:
+    # no description as it is not for local use
+    # order must be explicit as CRDs must be generated before header check
+    cmds:
+    - task: generate-crds
+    - task: ci-rest
+
+  ci-rest:
+    # these can all be run in parallel
+    deps: [header-check, test-cover, test-integration-envtest]
+
+  lint:
+    desc: Run lint checks.
+    cmds: [golangci-lint run -v]
+
+  lint-full:
+    desc: Run all lint checks.
+    cmds: [golangci-lint run -v --fast=false]
+
+  format-code:
+    desc: Ensure all code is correctly formatted.
+    cmds:
+    - gofmt -l -s -w {{.GENERATED_GO_DIRS_TO_TEST|catLines}}
+
+  tidy:
+    desc: Run 'go mod tidy'.
+    cmds: [go mod tidy]
+
+  generate-crds:
+    desc: Run controller-gen to generate CRD files.
+    cmds:
+    - find ./apis -type f -name "zz_generated.*" -delete
+    - controller-gen object:headerFile=../boilerplate.go.txt paths="./..."
+    - if [ -d "./config/crd/bases" ]; then find "./config/crd/bases" -type f -delete; fi
+    - controller-gen {{.CRD_OPTIONS}} rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
+    vars:
+      CRD_OPTIONS: "crd:crdVersions=v1,allowDangerousTypes=true"
+  
+  build:
+    desc: Generate the {{.GENERATED_APP}} binary.
+    deps: [tidy, lint]
+    cmds:
+    - CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -o ./bin/{{.GENERATED_APP}}
+
+  test:
+    desc: Run unit tests.
+    cmds:
+    - go test {{.GENERATED_GO_DIRS_TO_TEST|catLines}} -short -tags=noexit
+
+  test-cover:
+    desc: Run unit tests and output coverage.
+    cmds: 
+    - go test -short -tags=noexit -race -covermode atomic -coverprofile=cover.out -coverpkg="./..." {{.GENERATED_GO_DIRS_TO_TEST|catLines}}
+    - gcov2lcov -infile cover.out -outfile coverage.lcov
+
+  test-integration-envtest:
+    desc: Run integration tests with envtest and output coverage.
+    cmds:
+    # copy CRD files that are needed to a different folder, at the moment we cannot load all CRD files
+    - mkdir -p config/crd/bases/valid
+    - cp config/crd/bases/microsoft.batch.infra.azure.com_batchaccounts.yaml config/crd/bases/valid
+    - cp config/crd/bases/microsoft.storage.infra.azure.com_storageaccounts.yaml config/crd/bases/valid
+    - cp config/crd/bases/microsoft.resources.infra.azure.com_resourcegroups.yaml config/crd/bases/valid
+    # -race fails at the moment in controller-runtime
+    - go test -covermode atomic -coverprofile=cover-integration-envtest.out -coverpkg="./..." ./controllers/... -envtest 
+    - gcov2lcov -infile cover-integration-envtest.out -outfile coverage-integration-envtest.lcov
+  
+  cleanup-azure-resources:
+    desc: Removes any old resources created by tests.
+    cmds:
+      - |
+        rgs=`az group list --query '[*].{Name: name, CreatedAt: tags.CreatedAt}' \
```

Comment:
minor: Include comment as well?

> This finds all resource groups which match the specified pattern (k8sinfratest) and are older than a day
(86400 seconds). This is a bit horrible but it works...